neuralmonkey/attention/combination.py

```diff
@@ -169,8 +169,13 @@ def __init__(self,
             self.masks_concat = tf.concat(self._encoders_masks, 1)
     # pylint: enable=too-many-arguments
 
-    def initial_loop_state(self) -> AttentionLoopStateTA:
-        return empty_attention_loop_state()
+    @tensor
+    def batch_size(self) -> tf.Tensor:
+        # TODO: find a better solution
```

Comment:
Tohle je v pohodě solution, ale chtělo by to v konstruktoru kontrolovat, že mam aspoň jeden enkodér. A k tomu vyhodit warning když nebudou aspoň dva, jinak neni co kombinovat.

Reply:
Možná bych tu property zrušil a tu batch size získávál jen tam, kde je potřeba. Zvenku se k tomu nechodí a voláš to pokud vidim jednou.

Reply:
celkove se mi nelibi duplicita atributu batch_size, melo by to byt neco, co je globalne pristupne vsem model partum (kdyz je to definovane globalne v mainu)

Reply:
Myšlenka je, že model je (by měla bejt) funkce, který dáš data a zahraješ si za to nějakej tenzor. Kolik těch dat dáš by model neměl vůbec vědět, tj. batch size bude vždycky symbolickej tenzor a jakýkoliv řešení s batch size předávanou "zvenku" bude (nejspíš) špatný. Takže chceš mít někde symbolickou batch size. Tu si můžeš třeba udělat v placeholderu v base attentionu, do kterýho narveš ve `feed_dict`u `len(dataset)`. Jinak to musíš vyzevlit od ancestorů a ty to maj pokaždý jinde. Kór když jsi hierarchickej attention.

Reply:
To by konec konců mohl mít každej model part, podobně jako `train_mode`, který taky má skoro každej model part. Takže navrhuju udělat defaultní implementaci feed dictu v `ModelPart`, která nafeeduje `self.train_mode`  a `self.batch_size`.

Reply:
jop, souhlasim, ze by to asi mel byt atribut spolecneho predka, tedy ModelPart; mozna ani nemusi byt abstraktni